### PR TITLE
{ryujinx,yuzu-ea,yuzu-mainline}: bump

### LINKED ISSUE #128239: yuzu-mainline doesn't build on 21.05
**Describe the bug**

`yuzu-mainline` fails to build.

**To Reproduce**
1. add `yuzu-mainline` to `environment.systemPackages`.
2. `nixos-rebuild switch`
3. observe failure

**Expected behavior**

**Screenshots**

**Additional context**

most relevant part of build log:
```
[ERROR] fatal: git cat-file: could not get object info
[ERROR] Unrecognized git object type:
[ERROR] Unable to checkout eac2416b8fdb2cb9c867a538100bf95326bad75b from https://github.com/yuzu-emu/mbedtls.
[ERROR] builder for '/nix/store/dqa14b10hb9w2s4758v6zqhr9wyn9m5g-source.drv' failed with exit code 1
```

**Notify maintainers**

@IvarWithoutBones 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
builtins.fetchTarball {
  name = "nixos-stable-2021-06-25";
  url = "https://github.com/nixos/nixpkgs/archive/b72bde7c4a1f9c9bf1a161f0c267186ce3c6483c.tar.gz";
  sha256 = "1cihs542msfldcawcsk6s2bs8bpz82ldy6axsrax8ngw2gllqrcd";
}
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: yuzu-mainline
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
This updates all Nintendo Switch emulators currently in nixpkgs.

For yuzu, I've removed a patch that aren't required anymore, together with the CMake flags from said patch. I've changed the formatting of the title bar, as they made some changes upstream. I've also added the `dontFixCmake` flag, as it made the build fail. Some bug on their end broke fetching the source archive, as described in #128239. It seems this got fixed with later versions.

Fixes: #128239

For Ryujinx, I've hacked around https://github.com/Ryujinx/Ryujinx/issues/2349. Not sure if there is a better solution for this? Closure size doesn't seem to have changed that much, but it would be nice to have this be removed from the package.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129492` at 32b01d8c run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/84718ae402dec358041335eb86b75c4a">yuzu-ea</a></li>
    <li><a href="https://gist.github.com/de840591a21c807fe7c1cad0bf890c29">yuzu-mainline</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by IvarWithoutBones** [comment]
> Result of `nixpkgs-review pr 129492` at [32b01d8](https://github.com/NixOS/nixpkgs/commit/32b01d8c192e6838dee84c94305cf151c6d03de5) run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 2 packages failed to build:
> * [yuzu-ea](https://gist.github.com/84718ae402dec358041335eb86b75c4a)
> * [yuzu-mainline](https://gist.github.com/de840591a21c807fe7c1cad0bf890c29)
> 
> Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
> Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

yuzu currently does not support aarch64, so I marked it as broken on there.

File: pkgs/misc/emulators/ryujinx/default.nix [hunk 1]
```diff
@@ -79,9 +79,13 @@ in stdenv.mkDerivation rec {
       --output $out/lib/ryujinx
     shopt -s extglob
 
+    # TODO: fix this hack https://github.com/Ryujinx/Ryujinx/issues/2349
+    mkdir -p $out/lib/sndio-6
+    cp ${sndio}/lib/libsndio.so $out/lib/sndio-6/libsndio.so.6
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/ryujinx/default.nix:84]: Should we use ln here?

Comment (IvarWithoutBones) [reply pkgs/misc/emulators/ryujinx/default.nix]: Sounds good :+1:

**Comment by Mic92** [comment]
Ryujinx seems to not find pulseaudio:

```
$ Ryujinx
00:00:00.002 |I| Configuration LogValueChange: EnableFileLog set to: True
00:00:00.035 |I| Configuration LogValueChange: ResScale set to: 1
00:00:00.038 |I| Configuration LogValueChange: ResScaleCustom set to: 1
00:00:00.038 |I| Configuration LogValueChange: MaxAnisotropy set to: -1
00:00:00.040 |I| Configuration LogValueChange: AspectRatio set to: Fixed16x9
00:00:00.047 |I| Configuration LogValueChange: EnableDockedMode set to: True
00:00:00.070 |I| Configuration LogValueChange: EnableVsync set to: True
00:00:00.070 |I| Configuration LogValueChange: EnableShaderCache set to: True
00:00:00.071 |I| Configuration LogValueChange: EnablePtc set to: True
00:00:00.071 |I| Configuration LogValueChange: EnableFsIntegrityChecks set to: True
00:00:00.071 |I| Configuration LogValueChange: FsGlobalAccessLogMode set to: 0
00:00:00.072 |I| Configuration LogValueChange: AudioBackend set to: OpenAl
00:00:00.073 |I| Configuration LogValueChange: MemoryManagerMode set to: HostMappedUnsafe
00:00:00.073 |I| Configuration LogValueChange: ExpandRam set to: False
00:00:00.073 |I| Configuration LogValueChange: IgnoreMissingServices set to: False
00:00:00.304 |N| Application PrintSystemInfo: Ryujinx Version: 1.0.6954
00:00:00.316 |N| Application Print: Operating System: Linux 5.12.14 #1-NixOS SMP Tue Jan 1 00:00:00 UTC 1980 (X64)
00:00:00.316 |N| Application Print: CPU: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz ; 8 logical
00:00:00.316 |N| Application Print: RAM: Total 15805 MB ; Available 1383 MB
00:00:00.320 |N| Application PrintSystemInfo: Logs Enabled: Info, Warning, Error, Guest, Stub
00:00:00.320 |N| Application PrintSystemInfo: Launch Mode: UserProfile
00:00:03.641 |E| Application Initialize: SDL2 initlaization failed with error "Failed loading libpulse-simple.so.0: libpulse-simple.so.0: cannot open shared object file: No such file or directory"
00:00:03.681 |E| Application : Unhandled exception caught: System.Exception: SDL2 initlaization failed with error "Failed loading libpulse-simple.so.0: libpulse-simple.so.0: cannot open shared object file: No such file or directory"
   at Ryujinx.SDL2.Common.SDL2Driver.Initialize() in /build/source/Ryujinx.SDL2.Common/SDL2Driver.cs:line 69
   at Ryujinx.Input.SDL2.SDL2GamepadDriver..ctor() in /build/source/Ryujinx.Input.SDL2/SDL2GamepadDriver.cs:line 23
   at Ryujinx.Ui.MainWindow..ctor(Builder builder) in /build/source/Ryujinx/Ui/MainWindow.cs:line 259
   at Ryujinx.Program.Main(String[] args) in /build/source/Ryujinx/Program.cs:line 157
Unhandled exception. System.Exception: SDL2 initlaization failed with error "Failed loading libpulse-simple.so.0: libpulse-simple.so.0: cannot open shared object file: No such file or directory"
   at Ryujinx.SDL2.Common.SDL2Driver.Initialize() in /build/source/Ryujinx.SDL2.Common/SDL2Driver.cs:line 69
   at Ryujinx.Input.SDL2.SDL2GamepadDriver..ctor() in /build/source/Ryujinx.Input.SDL2/SDL2GamepadDriver.cs:line 23
   at Ryujinx.Ui.MainWindow..ctor(Builder builder) in /build/source/Ryujinx/Ui/MainWindow.cs:line 259
   at Ryujinx.Program.Main(String[] args) in /build/source/Ryujinx/Program.cs:line 157
Aborted (core dumped)
```

**Comment by IvarWithoutBones** [comment]
> Ryujinx seems to not find pulseaudio:
> 
> ```
> $ Ryujinx
> 00:00:00.002 |I| Configuration LogValueChange: EnableFileLog set to: True
> 00:00:00.035 |I| Configuration LogValueChange: ResScale set to: 1
> 00:00:00.038 |I| Configuration LogValueChange: ResScaleCustom set to: 1
> 00:00:00.038 |I| Configuration LogValueChange: MaxAnisotropy set to: -1
> 00:00:00.040 |I| Configuration LogValueChange: AspectRatio set to: Fixed16x9
> 00:00:00.047 |I| Configuration LogValueChange: EnableDockedMode set to: True
> 00:00:00.070 |I| Configuration LogValueChange: EnableVsync set to: True
> 00:00:00.070 |I| Configuration LogValueChange: EnableShaderCache set to: True
> 00:00:00.071 |I| Configuration LogValueChange: EnablePtc set to: True
> 00:00:00.071 |I| Configuration LogValueChange: EnableFsIntegrityChecks set to: True
> 00:00:00.071 |I| Configuration LogValueChange: FsGlobalAccessLogMode set to: 0
> 00:00:00.072 |I| Configuration LogValueChange: AudioBackend set to: OpenAl
> 00:00:00.073 |I| Configuration LogValueChange: MemoryManagerMode set to: HostMappedUnsafe
> 00:00:00.073 |I| Configuration LogValueChange: ExpandRam set to: False
> 00:00:00.073 |I| Configuration LogValueChange: IgnoreMissingServices set to: False
> 00:00:00.304 |N| Application PrintSystemInfo: Ryujinx Version: 1.0.6954
> 00:00:00.316 |N| Application Print: Operating System: Linux 5.12.14 #1-NixOS SMP Tue Jan 1 00:00:00 UTC 1980 (X64)
> 00:00:00.316 |N| Application Print: CPU: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz ; 8 logical
> 00:00:00.316 |N| Application Print: RAM: Total 15805 MB ; Available 1383 MB
> 00:00:00.320 |N| Application PrintSystemInfo: Logs Enabled: Info, Warning, Error, Guest, Stub
> 00:00:00.320 |N| Application PrintSystemInfo: Launch Mode: UserProfile
> 00:00:03.641 |E| Application Initialize: SDL2 initlaization failed with error "Failed loading libpulse-simple.so.0: libpulse-simple.so.0: cannot open shared object file: No such file or directory"
> 00:00:03.681 |E| Application : Unhandled exception caught: System.Exception: SDL2 initlaization failed with error "Failed loading libpulse-simple.so.0: libpulse-simple.so.0: cannot open shared object file: No such file or directory"
>    at Ryujinx.SDL2.Common.SDL2Driver.Initialize() in /build/source/Ryujinx.SDL2.Common/SDL2Driver.cs:line 69
>    at Ryujinx.Input.SDL2.SDL2GamepadDriver..ctor() in /build/source/Ryujinx.Input.SDL2/SDL2GamepadDriver.cs:line 23
>    at Ryujinx.Ui.MainWindow..ctor(Builder builder) in /build/source/Ryujinx/Ui/MainWindow.cs:line 259
>    at Ryujinx.Program.Main(String[] args) in /build/source/Ryujinx/Program.cs:line 157
> Unhandled exception. System.Exception: SDL2 initlaization failed with error "Failed loading libpulse-simple.so.0: libpulse-simple.so.0: cannot open shared object file: No such file or directory"
>    at Ryujinx.SDL2.Common.SDL2Driver.Initialize() in /build/source/Ryujinx.SDL2.Common/SDL2Driver.cs:line 69
>    at Ryujinx.Input.SDL2.SDL2GamepadDriver..ctor() in /build/source/Ryujinx.Input.SDL2/SDL2GamepadDriver.cs:line 23
>    at Ryujinx.Ui.MainWindow..ctor(Builder builder) in /build/source/Ryujinx/Ui/MainWindow.cs:line 259
>    at Ryujinx.Program.Main(String[] args) in /build/source/Ryujinx/Program.cs:line 157
> Aborted (core dumped)
> ```

Ah, this is due to the default audio configuration being set to `SDL2`, which needs pulseaudio. I changed that on my machine, so I never noticed the issue. Should be all fixed now, thanks for the report!

**Pull request merged**